### DISCUSSION TITLE
No status addresses after successful build and serving has started

### DISCUSSION THREAD

**Discussion by HanBrol0** [OpenFunction/OpenFunction#512] [Q&A]
Hello all,

I'm very interested in using OpenFunction, and have been fumbling my way to get it working in my 3-node cluster homelab.

I feel like I'm missing something obvious, but I'm struggling to get a function to start serving.  I'm able to get the build to succeed and the Serving State goes to 'Starting', but eventually ends in failure.  I get no output when I run (openfunction-demo is the name of the function):

kubectl get function openfunction-demo -o=jsonpath='{.status.addresses}'

Any ideas on where to look?  Maybe some logs I'm unaware of?  Any help would be great.

I am using the Go sample function to test.

Here is the ServingState:
![ServingState](https://github.com/OpenFunction/OpenFunction/assets/43140648/6b28f076-d102-4788-b116-4b5343cacd17)

kubectl describe function
![function-describe](https://github.com/OpenFunction/OpenFunction/assets/43140648/5be03b1a-816c-4f98-a6ce-5c49dbb0a7fc)

kubectl get function openfunction-demo -o=jsonpath='{.status.addresses}'
![NoAddresses](https://github.com/OpenFunction/OpenFunction/assets/43140648/c80b3b90-cfd3-438c-9509-2a6cba7bd161)

Thanks!

**Comment by benjaminhuo** [comment]
@wanjunlei @wrongerror any idea?

**Comment by wanjunlei** [comment]
@HanBrol0 The knative serving is not ready, please see the log of pod in the knative-serving namespace for more information.

**Reply by HanBrol0** [reply]
Thanks for the tip.  I've attached at log file with logs that are repeating once the build succeeds.  

I can confirm that the IP 10.111.119.27 maps to the ClusterIP of the  serving object created.  However, eventually the ServingState still enters a failed state.  The last two lines are taken after the State fails.

[knative-serving-activator-log.txt](https://github.com/OpenFunction/OpenFunction/files/13722675/knative-serving-activator-log.txt)
